xarray/backends/api.py

```diff
@@ -474,6 +470,15 @@ def open_dataset(
             "all other options must be passed as keyword arguments"
         )
 
+    # TODO remove after v0.19
+    if kwargs.pop("lock", None):
+        warnings.warn(
+            "The kwarg 'lock' has been deprecated, and is now "
+            "ignored. In the future passing lock will "
+            "raise an error.",
+            DeprecationWarning,
+        )
```

Comment:
should we include the target version in the warning and remove the comment?

Reply:
I did that originally but [Max said above](https://github.com/pydata/xarray/pull/5237#discussion_r623245556) that it might be better to leave it out... I think this is fine to be honest.

Reply:
hmm... in any case, I think it would be good to keep the error message for `open_dataset` and `open_dataarray` synchronized

Reply:
We could always add a specific date of deprecation later on, even if that would extend the cycle. It doesn't really matter if this check just sits here for a while I suppose.

Oh I didn't mean to make them different! Will fix now.